packages/server/controllers/events.js

```diff
@@ -47,6 +47,52 @@ async function getEvents(limit, action, eventId) {
   return events;
 }
 
+async function getPublicUpcomingEvents(limit, action, eventId) {
+  let events;
+
+  switch (action) {
+    case "next":
```

Comment:
Using an `action` parameter and a switch statement doesn't make sense in this case because we only have one "action", going forward for infinite scrolling, unlike the other route.

Instead, you can keep the Prisma queries but replace the switch statement and test if the cursor `eventId` is null.